server/model/monitor.js

```diff
@@ -395,19 +397,28 @@ class Monitor extends BeanModel {
                             }
                             const lastBeat = await Monitor.getPreviousHeartbeat(child.id);
 
-                            // Only change state if the monitor is in worse conditions then the ones before
+                            // Only change state if the monitor is in worse conditions than the ones before
                             // lastBeat.status could be null
                             if (!lastBeat) {
                                 bean.status = PENDING;
                             } else if (bean.status === UP && (lastBeat.status === PENDING || lastBeat.status === DOWN)) {
                                 bean.status = lastBeat.status;
+                                failChild = child.name;
                             } else if (bean.status === PENDING && lastBeat.status === DOWN) {
                                 bean.status = lastBeat.status;
+                                failChild = child.name;
+                            }
+                            if (lastBeat.status === DOWN) {
+                                failCount++;
                             }
                         }
 
                         if (bean.status !== UP) {
-                            bean.msg = "Child inaccessible";
+                            if (failCount <= 1) {
+                                bean.msg = "Child '" + failChild + "' inaccessible";
+                            } else {
+                                bean.msg = "Child '" + failChild + "' and " + (failCount - 1) + " others inaccessible";
```

Comment:
nit: please use templates instead of string concatination, as this is more reaable if including `'` and `"`

```suggestion
                                bean.msg = `Child '${failChild}' inaccessible`;
                            } else {
                                bean.msg = `Child '${failChild}' and ${failCount - 1} others inaccessible`;
```